external/loaders/tests/test__protocol.py

```diff
@@ -0,0 +1,25 @@
+from typing_extensions import runtime_checkable, Protocol
+
+
+def test_extended_kwargs_signature_is_still_instance():
```

Comment:
did you end up using the `Protocol` or the `Callable` for the Mappingfunction? If the latter, then these tests can be removed.